compat/maven-compat/src/main/java/org/apache/maven/repository/legacy/DefaultUpdateCheckManager.java

```diff
@@ -303,7 +301,7 @@ private Properties read(File touchfile) {
                 Properties props = new Properties();
 
                 try (FileInputStream in = new FileInputStream(touchfile)) {
-                    try (FileLock lock = in.getChannel().lock(0, Long.MAX_VALUE, true)) {
+                    try (FileLock ignored = in.getChannel().lock(0, Long.MAX_VALUE, true)) {
```

Comment:
It's not ignored. Is PMD simply wrong here?

Reply:
really? check out the usage. Its **unused for sure**, or the last 15 years i have done everything wrong, not even understood basics in pointer and reference.

https://www.geeksforgeeks.org/reference-variable-in-java/

Reply:
@desruisseaux point out correctly that its used `indirectly`, having the implizit `close()` call.

As IDEA and PMD both point this out is seems ok to rename to give proper context. Its ignored as handled by default.

Other way should be an inline comment for suppression.

Reply:
indeed. This is the only use, which is an exception of the normal pattern (using is explizit).

To make this clear the convention seems naming it `ignroed`, at least considering IDEA and PMD, both recognize the variable as marked explizit, as only used implicit.

Now using the resource which the statement is all about, seems like an error, thats why its pointes out by these tools.
Im altho wondering why its not used, allowing a comment to explain too.
Then the comment end up in the code again, to be self explaining.